backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/PresentationRequestHelper.java

```diff
@@ -0,0 +1,126 @@
+
+import java.util.*;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class PresentationRequestHelper {
+
+    /**
+     * Returns a valid PresentationRequest if possible (no preference)
+     * 
+     * @param presentationExchange {@link PresentationExchangeRecord}
+     * @param validCredentials     {@link List<PresentationRequestCredentials>}
+     * @return {@link PresentationRequest}
+     */
+    public static Optional<PresentationRequest> buildAny(
+            PresentationExchangeRecord presentationExchange,
+            List<PresentationRequestCredentials> validCredentials)
+            throws PresentationConstructionException {
+        Optional<PresentationRequest> result = Optional.empty();
+        result = Optional.of(PresentationRequest.builder()
+                .requestedAttributes(buildRequestedAttributes(
+                        presentationExchange.getPresentationRequest().getRequestedAttributes(),
+                        validCredentials))
+                .requestedPredicates(buildRequestedPredicates(
+                        presentationExchange.getPresentationRequest().getRequestedPredicates(),
+                        validCredentials))
+                .build());
+        return result;
+
+    }
+
+    // TODO update to handle any shape of RequestedAttributes
+    private static Map<String, PresentationRequest.IndyRequestedCredsRequestedAttr> buildRequestedAttributes(
+            Map<String, PresentProofRequest.ProofRequest.ProofAttributes> requestedAttributes,
+            List<PresentationRequestCredentials> validCredentials)
+            throws PresentationConstructionException {
+
+        Map<String, PresentationRequest.IndyRequestedCredsRequestedAttr> result = new HashMap<>();
+
+        for (Map.Entry<String, PresentProofRequest.ProofRequest.ProofAttributes> reqAttr : requestedAttributes
+                .entrySet()) {
+            List<JsonObject> restrictions = reqAttr.getValue().getRestrictions();
+            JsonElement restriction_schema_id = restrictions.get(0).get("schema_id");
+            JsonElement restriction_cred_def_id = restrictions.get(0).get("cred_def_id");
+
+            Optional<PresentationRequestCredentials> cred = Optional.empty();
+
+            if (restriction_schema_id != null) {
+                cred = validCredentials.stream()
+                        .filter(vc -> vc.getCredentialInfo().getSchemaId().equals(
+                                restriction_schema_id.getAsString()))
+                        .findFirst();
+            } else if (restriction_cred_def_id != null) {
+                cred = validCredentials.stream()
+                        .filter(vc -> vc.getCredentialInfo().getCredentialDefinitionId().equals(
+                                restriction_cred_def_id.getAsString()))
+                        .findFirst();
+            } else {
+                cred = validCredentials.stream().findFirst();
```

Comment:
If there is no schemaId and no credentialDefinitionId in the request and the request contains several sections you will answer with the same section multiple times

Reply:
What this will do is always grab the first 'valid credential' and use that to satisfy all `requestedAttributes`, and if that credential doesn't have the field that's being requested, aca-py will likely throw an error, but as long as they have `cred_def_id` or `schema_id` then the filtering will use the appropriate credential.